### ISSUE TITLE
Container stopped, but agent does not notice

### ISSUE THREAD

**Issue by fabfuel** [aws/amazon-ecs-agent#333]
The following container stopped:
```
CONTAINER ID        IMAGE                                                           COMMAND                  CREATED             STATUS                     PORTS                                              NAMES
f9a0e73a2ef7        XXXXXXX.dkr.ecr.us-east-1.amazonaws.com/jenkins/slave:latest    "/usr/local/bin/dind "   23 hours ago        Exited (255) 4 hours ago                                                      ecs-jenkins-slave-8-slave-8ae9f6859ac480fc9401
```

But the agent says it's still `RUNNING`.

Even if I manually stop it, the task is shown under the "Stopped" tasks, but the "Last Status" is still `RUNNING (Task stopped by user)`.

Biggest issue, the service creates a new task, after I manually stopped the old one, but there is no new container started and the new task stays in status `PENDING`.

Any advice?
Can I provide any more information?
The ecs-agent.log shows some warnings, but nothing in particular with the mentioned container id.

Best
Fabian

**Comment by fabfuel** [comment]
Workaround: 
I manually restarted the the `ecs-agent` container and after that everything was fixed. The old task changed to status `STOPPED (Task stopped by user)` and the new task started a new container and switched to status `RUNNING`.

**Comment by juanrhenals** [comment]
Hello @fabfuel, we'd like to help but we'll need some more information from you.

Some questions:
1) Is this a recurring issue for you? If so, do you have a consistent repro for the issue that you can include?
2) Are you are using the ECS-optimized AMI? If so, which version?
3) What Docker version are you using?
4) What ECS Agent version are you using?
5) Can you provide the ECS Agent logs and the contents of /var/log/docker?

We have a troubleshooting page [here](http://docs.aws.amazon.com/AmazonECS/latest/developerguide/troubleshooting.html) that may help you as well.

Thanks!

**Comment by lafraia** [comment]
I'm having the same issue running `ami-60b6c60a` (Amazon oficial AMI - not ECS-optimized though). After restarting ecs-agent container, it works. After checking ecs-agent logs, I see some issues reading CPU info (might be related to #174). The same behaviour happens on updating cluster's services/task definition.

```
# docker logs ecs-agent
2016-03-11T17:59:24Z [INFO] Starting Agent: Amazon ECS Agent - v1.8.1 (304495b)
2016-03-11T17:59:24Z [INFO] Loading configuration
2016-03-11T17:59:24Z [WARN] Invalid value for task cleanup duration, will be overridden to 3h0m0s module="config" parsed value="0" minimum threshold="1m0s"
2016-03-11T17:59:24Z [INFO] Checkpointing is enabled. Attempting to load state
2016-03-11T17:59:24Z [INFO] Loading state! module="statemanager"
2016-03-11T17:59:24Z [INFO] Restored cluster 'Admin'
2016-03-11T17:59:24Z [INFO] Detected Docker versions [1.17 1.18 1.19 1.20]
2016-03-11T17:59:24Z [INFO] Restored from checkpoint file. I am running as 'arn:aws:ecs:us-east-1:xxxxxxxxxxxxxx:container-instance/51706a42-1a49-4cad-a53d-246107cbfe2b' in cluster 'xxxxx'
2016-03-11T17:59:24Z [INFO] Registered! module="api client"
2016-03-11T17:59:24Z [INFO] Saving state! module="statemanager"
2016-03-11T17:59:24Z [INFO] Beginning Polling for updates
2016-03-11T17:59:24Z [INFO] Initializing stats engine module="stats"
2016-03-11T17:59:25Z [INFO] Creating poll dialer module="ws client" host="ecs-a-1.us-east-1.amazonaws.com"
2016-03-11T17:59:25Z [INFO] Creating poll dialer module="ws client" host="ecs-t-1.us-east-1.amazonaws.com"
2016-03-11T17:59:25Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:217c3763e11b1169eb42d4900232382e3e9d090d94ee61b2537733079d047b08}"
2016-03-11T17:59:25Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:b87342baf0936b36ddf34577a574830f13d90bea86f3de289c5e3a8a2f66f047}"
2016-03-11T17:59:25Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:badffc49c44ffb1496ea37c6bef95054508c419e1546e30334b1fc8af71e0442}"
2016-03-11T17:59:25Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:a5510d85f5a34451d2611f8c2d768c600ccabb96f54885024d16ea1c9542e142}"
2016-03-11T17:59:25Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:3c275f81fe3f0a0961d13f3cfb012a3cac8528c501b8ea02834b312cd15e0d20}"
2016-03-11T17:59:25Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:c4aee01c2ba32f26b091682cd2c80a7ec371389b6222b04017d82ed631aa9e5d}"
2016-03-11T17:59:25Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:5eb0db7d242c9267791b02b3eaea491dbd4331ca1f989750fc6c593032942906}"
2016-03-11T17:59:25Z [WARN] Error getting instance metrics module="tcs client" err="No task metrics to report"
2016-03-11T17:59:34Z [INFO] Saving state! module="statemanager"
```

**Comment by fabfuel** [comment]
Hi @juanrhenals 

1) It just happened the third time, but unfortunately I do not have a path how to reproduce it.
2) Yes, I'm using the ECS optimized AMI: amzn-ami-2015.09.e-amazon-ecs-optimized (ami-c3253caf)
3) Docker version:
```
Client:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5/1.9.1
 Built:
 OS/Arch:      linux/amd64

Server:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5/1.9.1
 Built:
 OS/Arch:      linux/amd64
```
4) ECS agent version is 1.8.0 (I just started the agent update to version 1.8.1, maybe that solves that issue. I'll keep you posted.)
5) ECS agent log:
```
2016-03-12T11:15:50Z [INFO] Creating poll dialer module="ws client" host="ecs-a-1.eu-central-1.amazonaws.com"
2016-03-12T11:18:18Z [INFO] Creating poll dialer module="ws client" host="ecs-t-1.eu-central-1.amazonaws.com"
2016-03-12T11:18:18Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:3d429371c484cd5030c190d80d420c0f6377949f1cc6fd2e1448bddb0fa90715}"
2016-03-12T11:18:18Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:ad133dd53c55dd07e9d2b552116d5b1baa2b1da7ad69e829b2fd4d2d2599db1b}"
2016-03-12T11:18:18Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:10ff6dd7a0fc9d7acee72121314d565d17608a39434dc60f53264e017542bc61}"
2016-03-12T11:18:18Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:78ea5ce65b7ff06f61451865b10d793ad448d06eddb374164765f39f7062e716}"
2016-03-12T11:18:18Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:fed3a833f051b233241f7097faba2a38ba307b34e6667d7a39b44e883dd8f030}"
2016-03-12T11:18:18Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:d60858c72806261309b2ed5d8713a00dced24ffcc229755eb6a5ae7dd7b69812}"
2016-03-12T11:18:18Z [WARN] Error getting instance metrics module="tcs client" err="No task metrics to report"
2016-03-12T11:28:18Z [INFO] Creating poll dialer module="ws client" host="ecs-t-1.eu-central-1.amazonaws.com"
2016-03-12T11:28:18Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:fed3a833f051b233241f7097faba2a38ba307b34e6667d7a39b44e883dd8f030}"
2016-03-12T11:28:18Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:d60858c72806261309b2ed5d8713a00dced24ffcc229755eb6a5ae7dd7b69812}"
2016-03-12T11:28:18Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:ad133dd53c55dd07e9d2b552116d5b1baa2b1da7ad69e829b2fd4d2d2599db1b}"
2016-03-12T11:28:18Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:10ff6dd7a0fc9d7acee72121314d565d17608a39434dc60f53264e017542bc61}"
2016-03-12T11:28:18Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:78ea5ce65b7ff06f61451865b10d793ad448d06eddb374164765f39f7062e716}"
2016-03-12T11:28:18Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:3d429371c484cd5030c190d80d420c0f6377949f1cc6fd2e1448bddb0fa90715}"
2016-03-12T11:28:18Z [WARN] Error getting instance metrics module="tcs client" err="No task metrics to report"
2016-03-12T11:43:55Z [INFO] Creating poll dialer module="ws client" host="ecs-a-1.eu-central-1.amazonaws.com"
2016-03-12T11:55:18Z [INFO] Creating poll dialer module="ws client" host="ecs-t-1.eu-central-1.amazonaws.com"
2016-03-12T11:55:18Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:3d429371c484cd5030c190d80d420c0f6377949f1cc6fd2e1448bddb0fa90715}"
2016-03-12T11:55:18Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:ad133dd53c55dd07e9d2b552116d5b1baa2b1da7ad69e829b2fd4d2d2599db1b}"
2016-03-12T11:55:18Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:10ff6dd7a0fc9d7acee72121314d565d17608a39434dc60f53264e017542bc61}"
2016-03-12T11:55:18Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:78ea5ce65b7ff06f61451865b10d793ad448d06eddb374164765f39f7062e716}"
2016-03-12T11:55:18Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:fed3a833f051b233241f7097faba2a38ba307b34e6667d7a39b44e883dd8f030}"
2016-03-12T11:55:18Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:d60858c72806261309b2ed5d8713a00dced24ffcc229755eb6a5ae7dd7b69812}"
2016-03-12T11:55:18Z [WARN] Error getting instance metrics module="tcs client" err="No task metrics to report"
2016-03-12T12:11:18Z [INFO] Creating poll dialer module="ws client" host="ecs-t-1.eu-central-1.amazonaws.com"
2016-03-12T12:11:18Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:10ff6dd7a0fc9d7acee72121314d565d17608a39434dc60f53264e017542bc61}"
2016-03-12T12:11:18Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:ad133dd53c55dd07e9d2b552116d5b1baa2b1da7ad69e829b2fd4d2d2599db1b}"
2016-03-12T12:11:18Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:78ea5ce65b7ff06f61451865b10d793ad448d06eddb374164765f39f7062e716}"
2016-03-12T12:11:18Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:fed3a833f051b233241f7097faba2a38ba307b34e6667d7a39b44e883dd8f030}"
2016-03-12T12:11:18Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:d60858c72806261309b2ed5d8713a00dced24ffcc229755eb6a5ae7dd7b69812}"
2016-03-12T12:11:18Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:3d429371c484cd5030c190d80d420c0f6377949f1cc6fd2e1448bddb0fa90715}"
2016-03-12T12:11:18Z [WARN] Error getting instance metrics module="tcs client" err="No task metrics to report"
2016-03-12T12:13:02Z [INFO] Creating poll dialer module="ws client" host="ecs-a-1.eu-central-1.amazonaws.com"
2016-03-12T12:13:02Z [INFO] Creating poll dialer module="ws client" host="ecs-a-1.eu-central-1.amazonaws.com"
2016-03-12T12:13:18Z [ERROR] Error getting message from ws backend module="ws client" err="unexpected EOF"
2016-03-12T12:13:18Z [INFO] Error from tcs; backing off module="tcs handler" err="unexpected EOF"
2016-03-12T12:13:19Z [INFO] Creating poll dialer module="ws client" host="ecs-t-1.eu-central-1.amazonaws.com"
2016-03-12T12:17:38Z [INFO] Removing container module="TaskEngine" task="testing:49 arn:aws:ecs:eu-central-1:172561589225:task/a5950357-f680-4da7-a556-4f91bf49dc08, Status: (STOPPED->STOPPED) Containers: [nginx (STOPPED->STOPPED),]" container="nginx(nginx:1.9) (STOPPED->STOPPED)"
2016-03-12T12:32:19Z [INFO] Creating poll dialer module="ws client" host="ecs-t-1.eu-central-1.amazonaws.com"
2016-03-12T12:32:19Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:ad133dd53c55dd07e9d2b552116d5b1baa2b1da7ad69e829b2fd4d2d2599db1b}"
2016-03-12T12:32:19Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:10ff6dd7a0fc9d7acee72121314d565d17608a39434dc60f53264e017542bc61}"
2016-03-12T12:32:19Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:78ea5ce65b7ff06f61451865b10d793ad448d06eddb374164765f39f7062e716}"
2016-03-12T12:32:19Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:fed3a833f051b233241f7097faba2a38ba307b34e6667d7a39b44e883dd8f030}"
2016-03-12T12:32:19Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:d60858c72806261309b2ed5d8713a00dced24ffcc229755eb6a5ae7dd7b69812}"
2016-03-12T12:32:19Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:3d429371c484cd5030c190d80d420c0f6377949f1cc6fd2e1448bddb0fa90715}"
2016-03-12T12:32:19Z [WARN] Error getting instance metrics module="tcs client" err="No task metrics to report"
2016-03-12T12:40:37Z [INFO] Creating poll dialer module="ws client" host="ecs-a-1.eu-central-1.amazonaws.com"
2016-03-12T12:42:20Z [INFO] Creating poll dialer module="ws client" host="ecs-t-1.eu-central-1.amazonaws.com"
2016-03-12T12:42:20Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:fed3a833f051b233241f7097faba2a38ba307b34e6667d7a39b44e883dd8f030}"
2016-03-12T12:42:20Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:d60858c72806261309b2ed5d8713a00dced24ffcc229755eb6a5ae7dd7b69812}"
2016-03-12T12:42:20Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:ad133dd53c55dd07e9d2b552116d5b1baa2b1da7ad69e829b2fd4d2d2599db1b}"
2016-03-12T12:42:20Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:10ff6dd7a0fc9d7acee72121314d565d17608a39434dc60f53264e017542bc61}"
2016-03-12T12:42:20Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:78ea5ce65b7ff06f61451865b10d793ad448d06eddb374164765f39f7062e716}"
2016-03-12T12:42:20Z [WARN] Error getting cpu stats module="stats" err="No data in the queue" container="&{DockerID:3d429371c484cd5030c190d80d420c0f6377949f1cc6fd2e1448bddb0fa90715}"
2016-03-12T12:42:20Z [WARN] Error getting instance metrics module="tcs client" err="No task metrics to report"
```

And the `/var/log/docker`
```
time="2016-03-12T10:59:54.236215651Z" level=info msg="GET /v1.17/containers/3d429371c484cd5030c190d80d420c0f6377949f1cc6fd2e1448bddb0fa90715/json"
time="2016-03-12T11:00:13.468035359Z" level=info msg="GET /v1.17/containers/d60858c72806261309b2ed5d8713a00dced24ffcc229755eb6a5ae7dd7b69812/json"
time="2016-03-12T11:08:01.837028005Z" level=info msg="GET /v1.17/containers/10ff6dd7a0fc9d7acee72121314d565d17608a39434dc60f53264e017542bc61/json"
time="2016-03-12T11:08:47.340851924Z" level=info msg="DELETE /v1.17/containers/efb79a484509163911e9ac3bfe087e529091bfe6a495264c02ac0681918f656e?v=1"
time="2016-03-12T11:08:47.368856698Z" level=info msg="GET /v1.17/containers/efb79a484509163911e9ac3bfe087e529091bfe6a495264c02ac0681918f656e/json"
time="2016-03-12T11:08:47.368995720Z" level=error msg="Handler for GET /v1.17/containers/efb79a484509163911e9ac3bfe087e529091bfe6a495264c02ac0681918f656e/json returned error: no such id: efb79a484509163911e9ac3bfe087e529091bfe6a495264c02ac0681918f656e"
time="2016-03-12T11:08:47.369024547Z" level=error msg="HTTP Error" err="no such id: efb79a484509163911e9ac3bfe087e529091bfe6a495264c02ac0681918f656e" statusCode=404
time="2016-03-12T11:08:58.352513301Z" level=info msg="GET /v1.17/containers/78ea5ce65b7ff06f61451865b10d793ad448d06eddb374164765f39f7062e716/json"
time="2016-03-12T11:08:58.566934276Z" level=info msg="DELETE /v1.17/containers/7686c22e86ce437d8802186ff9f8825fa0e67d2e55a54e22121c329284972230?v=1"
time="2016-03-12T11:08:58.568361272Z" level=info msg="DELETE /v1.17/containers/5b5d5456545d4661aef412b3c46af77cff7d1a170d2b8b4d2c550a7126264705?v=1"
time="2016-03-12T11:08:58.596205919Z" level=info msg="GET /v1.17/containers/7686c22e86ce437d8802186ff9f8825fa0e67d2e55a54e22121c329284972230/json"
time="2016-03-12T11:08:58.596373076Z" level=error msg="Handler for GET /v1.17/containers/7686c22e86ce437d8802186ff9f8825fa0e67d2e55a54e22121c329284972230/json returned error: no such id: 7686c22e86ce437d8802186ff9f8825fa0e67d2e55a54e22121c329284972230"
time="2016-03-12T11:08:58.596402809Z" level=error msg="HTTP Error" err="no such id: 7686c22e86ce437d8802186ff9f8825fa0e67d2e55a54e22121c329284972230" statusCode=404
time="2016-03-12T11:08:58.618511312Z" level=info msg="GET /v1.17/containers/5b5d5456545d4661aef412b3c46af77cff7d1a170d2b8b4d2c550a7126264705/json"
time="2016-03-12T11:08:58.618667535Z" level=error msg="Handler for GET /v1.17/containers/5b5d5456545d4661aef412b3c46af77cff7d1a170d2b8b4d2c550a7126264705/json returned error: no such id: 5b5d5456545d4661aef412b3c46af77cff7d1a170d2b8b4d2c550a7126264705"
time="2016-03-12T11:08:58.618701862Z" level=error msg="HTTP Error" err="no such id: 5b5d5456545d4661aef412b3c46af77cff7d1a170d2b8b4d2c550a7126264705" statusCode=404
time="2016-03-12T11:09:07.829667251Z" level=info msg="DELETE /v1.17/containers/ec42ba47c05ee94a70be58561e5127355d9967fb0d7eef75f11a85595e503636?v=1"
time="2016-03-12T11:09:07.851672841Z" level=info msg="GET /v1.17/containers/ec42ba47c05ee94a70be58561e5127355d9967fb0d7eef75f11a85595e503636/json"
time="2016-03-12T11:09:07.851828906Z" level=error msg="Handler for GET /v1.17/containers/ec42ba47c05ee94a70be58561e5127355d9967fb0d7eef75f11a85595e503636/json returned error: no such id: ec42ba47c05ee94a70be58561e5127355d9967fb0d7eef75f11a85595e503636"
time="2016-03-12T11:09:07.851858469Z" level=error msg="HTTP Error" err="no such id: ec42ba47c05ee94a70be58561e5127355d9967fb0d7eef75f11a85595e503636" statusCode=404
time="2016-03-12T11:09:34.701022178Z" level=info msg="GET /v1.17/containers/fed3a833f051b233241f7097faba2a38ba307b34e6667d7a39b44e883dd8f030/json"
time="2016-03-12T11:09:54.236449076Z" level=info msg="GET /v1.17/containers/3d429371c484cd5030c190d80d420c0f6377949f1cc6fd2e1448bddb0fa90715/json"
time="2016-03-12T11:10:13.468246829Z" level=info msg="GET /v1.17/containers/d60858c72806261309b2ed5d8713a00dced24ffcc229755eb6a5ae7dd7b69812/json"
time="2016-03-12T11:11:47.495930516Z" level=info msg="DELETE /v1.17/containers/48bac2543f40c20f06fb7ad2529e5579d9631477218fe40e050259b39b2a9b24?v=1"
time="2016-03-12T11:11:47.497381479Z" level=info msg="DELETE /v1.17/containers/5aa2305704496a3918f1ec157b7a7918b40541f61eacc77c6876cc3d1bcaf05c?v=1"
time="2016-03-12T11:11:47.509631685Z" level=info msg="DELETE /v1.17/containers/d13d0c76b82e893a1858a93d7ea00714af4d5470c80e977427afaa4ce7189c65?v=1"
time="2016-03-12T11:11:47.513740042Z" level=info msg="DELETE /v1.17/containers/ac7db40ea083646e4ac9dd768c53597e7490025dbac7110c143b60cd2f2ca60b?v=1"
time="2016-03-12T11:11:47.529549245Z" level=info msg="DELETE /v1.17/containers/de95da1acfc9893b3ef2dc048adae660908f959d49e0063181268a78dd76783d?v=1"
time="2016-03-12T11:11:47.532810950Z" level=info msg="GET /v1.17/containers/48bac2543f40c20f06fb7ad2529e5579d9631477218fe40e050259b39b2a9b24/json"
time="2016-03-12T11:11:47.536129212Z" level=error msg="Handler for GET /v1.17/containers/48bac2543f40c20f06fb7ad2529e5579d9631477218fe40e050259b39b2a9b24/json returned error: no such id: 48bac2543f40c20f06fb7ad2529e5579d9631477218fe40e050259b39b2a9b24"
time="2016-03-12T11:11:47.536159737Z" level=error msg="HTTP Error" err="no such id: 48bac2543f40c20f06fb7ad2529e5579d9631477218fe40e050259b39b2a9b24" statusCode=404
time="2016-03-12T11:11:47.550695343Z" level=info msg="GET /v1.17/containers/d13d0c76b82e893a1858a93d7ea00714af4d5470c80e977427afaa4ce7189c65/json"
time="2016-03-12T11:11:47.550811893Z" level=error msg="Handler for GET /v1.17/containers/d13d0c76b82e893a1858a93d7ea00714af4d5470c80e977427afaa4ce7189c65/json returned error: no such id: d13d0c76b82e893a1858a93d7ea00714af4d5470c80e977427afaa4ce7189c65"
time="2016-03-12T11:11:47.550839453Z" level=error msg="HTTP Error" err="no such id: d13d0c76b82e893a1858a93d7ea00714af4d5470c80e977427afaa4ce7189c65" statusCode=404
time="2016-03-12T11:11:47.567121212Z" level=info msg="GET /v1.17/containers/ac7db40ea083646e4ac9dd768c53597e7490025dbac7110c143b60cd2f2ca60b/json"
time="2016-03-12T11:11:47.567225889Z" level=error msg="Handler for GET /v1.17/containers/ac7db40ea083646e4ac9dd768c53597e7490025dbac7110c143b60cd2f2ca60b/json returned error: no such id: ac7db40ea083646e4ac9dd768c53597e7490025dbac7110c143b60cd2f2ca60b"
time="2016-03-12T11:11:47.567253923Z" level=error msg="HTTP Error" err="no such id: ac7db40ea083646e4ac9dd768c53597e7490025dbac7110c143b60cd2f2ca60b" statusCode=404
time="2016-03-12T11:11:47.575530711Z" level=info msg="GET /v1.17/containers/5aa2305704496a3918f1ec157b7a7918b40541f61eacc77c6876cc3d1bcaf05c/json"
time="2016-03-12T11:11:47.575634057Z" level=error msg="Handler for GET /v1.17/containers/5aa2305704496a3918f1ec157b7a7918b40541f61eacc77c6876cc3d1bcaf05c/json returned error: no such id: 5aa2305704496a3918f1ec157b7a7918b40541f61eacc77c6876cc3d1bcaf05c"
time="2016-03-12T11:11:47.575666466Z" level=error msg="HTTP Error" err="no such id: 5aa2305704496a3918f1ec157b7a7918b40541f61eacc77c6876cc3d1bcaf05c" statusCode=404
time="2016-03-12T11:11:47.592913678Z" level=info msg="GET /v1.17/containers/de95da1acfc9893b3ef2dc048adae660908f959d49e0063181268a78dd76783d/json"
time="2016-03-12T11:11:47.593049181Z" level=error msg="Handler for GET /v1.17/containers/de95da1acfc9893b3ef2dc048adae660908f959d49e0063181268a78dd76783d/json returned error: no such id: de95da1acfc9893b3ef2dc048adae660908f959d49e0063181268a78dd76783d"
time="2016-03-12T11:11:47.593078370Z" level=error msg="HTTP Error" err="no such id: de95da1acfc9893b3ef2dc048adae660908f959d49e0063181268a78dd76783d" statusCode=404
time="2016-03-12T11:11:47.796919212Z" level=info msg="DELETE /v1.17/containers/b596a69a3ea70140ac84bf0b636f6c01144c1ccfe9bf124d3a06ba3038d9ea4b?v=1"
time="2016-03-12T11:11:47.821803899Z" level=info msg="GET /v1.17/containers/b596a69a3ea70140ac84bf0b636f6c01144c1ccfe9bf124d3a06ba3038d9ea4b/json"
time="2016-03-12T11:11:47.821936512Z" level=error msg="Handler for GET /v1.17/containers/b596a69a3ea70140ac84bf0b636f6c01144c1ccfe9bf124d3a06ba3038d9ea4b/json returned error: no such id: b596a69a3ea70140ac84bf0b636f6c01144c1ccfe9bf124d3a06ba3038d9ea4b"
time="2016-03-12T11:11:47.821965000Z" level=error msg="HTTP Error" err="no such id: b596a69a3ea70140ac84bf0b636f6c01144c1ccfe9bf124d3a06ba3038d9ea4b" statusCode=404
time="2016-03-12T11:11:56.731133210Z" level=info msg="DELETE /v1.17/containers/53ae1c12e9592c6ecfd050325761cf01181686dc372f42da88b7b3a960d051ec?v=1"
time="2016-03-12T11:11:56.732722621Z" level=info msg="DELETE /v1.17/containers/28208f8847c73f2d0718a9c875b55101a7bec397600017162f5fc7a2e7a7e50e?v=1"
time="2016-03-12T11:11:56.762419203Z" level=info msg="GET /v1.17/containers/53ae1c12e9592c6ecfd050325761cf01181686dc372f42da88b7b3a960d051ec/json"
time="2016-03-12T11:11:56.762542053Z" level=error msg="Handler for GET /v1.17/containers/53ae1c12e9592c6ecfd050325761cf01181686dc372f42da88b7b3a960d051ec/json returned error: no such id: 53ae1c12e9592c6ecfd050325761cf01181686dc372f42da88b7b3a960d051ec"
time="2016-03-12T11:11:56.762570572Z" level=error msg="HTTP Error" err="no such id: 53ae1c12e9592c6ecfd050325761cf01181686dc372f42da88b7b3a960d051ec" statusCode=404
time="2016-03-12T11:11:56.782148258Z" level=info msg="GET /v1.17/containers/28208f8847c73f2d0718a9c875b55101a7bec397600017162f5fc7a2e7a7e50e/json"
time="2016-03-12T11:11:56.782276879Z" level=error msg="Handler for GET /v1.17/containers/28208f8847c73f2d0718a9c875b55101a7bec397600017162f5fc7a2e7a7e50e/json returned error: no such id: 28208f8847c73f2d0718a9c875b55101a7bec397600017162f5fc7a2e7a7e50e"
time="2016-03-12T11:11:56.782305038Z" level=error msg="HTTP Error" err="no such id: 28208f8847c73f2d0718a9c875b55101a7bec397600017162f5fc7a2e7a7e50e" statusCode=404
time="2016-03-12T11:13:42.782844278Z" level=info msg="DELETE /v1.17/containers/dac60f5708239b3884e516ff2dc8c812b0e7c3764b62d854384c7cbce7bbdb42?v=1"
time="2016-03-12T11:13:42.812349752Z" level=info msg="GET /v1.17/containers/dac60f5708239b3884e516ff2dc8c812b0e7c3764b62d854384c7cbce7bbdb42/json"
time="2016-03-12T11:13:42.812496486Z" level=error msg="Handler for GET /v1.17/containers/dac60f5708239b3884e516ff2dc8c812b0e7c3764b62d854384c7cbce7bbdb42/json returned error: no such id: dac60f5708239b3884e516ff2dc8c812b0e7c3764b62d854384c7cbce7bbdb42"
time="2016-03-12T11:13:42.812526572Z" level=error msg="HTTP Error" err="no such id: dac60f5708239b3884e516ff2dc8c812b0e7c3764b62d854384c7cbce7bbdb42" statusCode=404
time="2016-03-12T11:14:04.389930092Z" level=info msg="DELETE /v1.17/containers/3a5be4a3748f8b3956fc41c4f6e48b76a5362e444158145d0c1f581f70434121?v=1"
time="2016-03-12T11:14:04.419581557Z" level=info msg="GET /v1.17/containers/3a5be4a3748f8b3956fc41c4f6e48b76a5362e444158145d0c1f581f70434121/json"
time="2016-03-12T11:14:04.419708523Z" level=error msg="Handler for GET /v1.17/containers/3a5be4a3748f8b3956fc41c4f6e48b76a5362e444158145d0c1f581f70434121/json returned error: no such id: 3a5be4a3748f8b3956fc41c4f6e48b76a5362e444158145d0c1f581f70434121"
time="2016-03-12T11:14:04.419742946Z" level=error msg="HTTP Error" err="no such id: 3a5be4a3748f8b3956fc41c4f6e48b76a5362e444158145d0c1f581f70434121" statusCode=404
time="2016-03-12T11:15:50.500514957Z" level=info msg="GET /v1.17/version"
time="2016-03-12T11:18:01.837240045Z" level=info msg="GET /v1.17/containers/10ff6dd7a0fc9d7acee72121314d565d17608a39434dc60f53264e017542bc61/json"
time="2016-03-12T11:18:58.352764308Z" level=info msg="GET /v1.17/containers/78ea5ce65b7ff06f61451865b10d793ad448d06eddb374164765f39f7062e716/json"
time="2016-03-12T11:19:34.701316008Z" level=info msg="GET /v1.17/containers/fed3a833f051b233241f7097faba2a38ba307b34e6667d7a39b44e883dd8f030/json"
time="2016-03-12T11:19:54.236672331Z" level=info msg="GET /v1.17/containers/3d429371c484cd5030c190d80d420c0f6377949f1cc6fd2e1448bddb0fa90715/json"
time="2016-03-12T11:20:13.468464550Z" level=info msg="GET /v1.17/containers/d60858c72806261309b2ed5d8713a00dced24ffcc229755eb6a5ae7dd7b69812/json"
time="2016-03-12T11:28:01.837510101Z" level=info msg="GET /v1.17/containers/10ff6dd7a0fc9d7acee72121314d565d17608a39434dc60f53264e017542bc61/json"
time="2016-03-12T11:28:58.353007186Z" level=info msg="GET /v1.17/containers/78ea5ce65b7ff06f61451865b10d793ad448d06eddb374164765f39f7062e716/json"
time="2016-03-12T11:29:34.701486803Z" level=info msg="GET /v1.17/containers/fed3a833f051b233241f7097faba2a38ba307b34e6667d7a39b44e883dd8f030/json"
```

Thanks
Fabian

**Comment by samuelkarp** [comment]
@fabfuel Were you able to test with the 1.8.1 agent?

**Comment by fabfuel** [comment]
Hi @samuelkarp,

the agent update did not work (it always failed). But yesterday I recreated the cluster with the current AMI and *so far* did issue did not appear, yet. 

Best
Fabian

**Comment by samuelkarp** [comment]
Thanks for letting me know.  We had a deadlock in 1.8.0 that we fixed with [1.8.1](https://github.com/aws/amazon-ecs-agent/releases/tag/v1.8.1).  The deadlock likely would have prevented the update (using either the `UpdateContainerAgent` API or the console) from working, but updating the `ecs-init` package on the instance or launching the new AMI should work.

Please let us know if the 1.8.1 agent resolves your issue or if you experience problems with it.  I'll leave this open for a few more days or until you let us know that it has resolved your issue.

**Comment by samuelkarp** [comment]
@fabfuel I'm going to close this issue for now since I haven't heard back from you.  If you continue to experience problems, please let us know.
